### ISSUE TITLE
Finger Length Slider in Avatar VRIK

### ISSUE THREAD

**Issue by GearBell** [Neos-Metaverse/NeosPublic#2757]
### Is your tweak request related to a problem? Please describe.

In the VRIK for avatar settings there is a slider to lengthen or shorten the arms without causing deformities to thier scale/rotations. Theres been a few instances where Ive realized I'd like to make the avatars fingers longer as well, but unfortunatly realizing that after Ive loaded logix into the avatar - rig transfer tools (for my avatars) simply do not work. Just like the MLP Arm Length slider, what Im requesting the Finger Length do is put a wider gap in between each finger rather than scale them, so they do not deform when rotating.

### Describe what would you like tweaked

> I am requesting an additional slider in the VRIK component called Finger Length, to affect the gap distance between each finger bone.

### Describe alternatives you've considered

Simply scaling the fingers deforms them horribly, and the only other alternative is redoing the avatar from scratch, simply because the finger length seems too short.

**Comment by H3BO3** [comment]
Wouldn't this be in the hand poser, not VRIK?

**Comment by GearBell** [comment]
Actually thats a good point... doesnt matter where it is, Id like to have the slider.

**Comment by H3BO3** [comment]
Hmm... I seem to be unable to edit issue titles anymore. Unsure how recent this is, but I would change it if I could.

**Comment by GearBell** [comment]
fixed!

**Comment by Earthmark** [comment]
I don't think this is hand poser related directly, as the hand poser works on the rotations but does nothing to the positions. Perhaps an external tool can be used that multiplies the transform of each bone by a fixed scalar? That would give the desired effect I think.

Also, this may be related to #2489

**Comment by H3BO3** [comment]
Affecting scale transforms affects all scales of children and leads to massive weirdness. This is a pretty good candidate for the hand poser. The whole point is so the hand poser can start controlling positions. @Earthmark

**Comment by Earthmark** [comment]
Transform, not scale, transform is the offset vector. Each slot has a trs (transform,

**Comment by H3BO3** [comment]
Realized I misread scalar from your post as well

**Comment by Earthmark** [comment]
If the hand poser isn't getting data about the users finger length then I do not agree, if the hand data sources only work on providing rotation information then I feel they should only bind rotation information. Adding bindings for translation can be done easily enough with an added lerp component (although it's a bit painful to add a component to each joint).

**Comment by Earthmark** [comment]
A one off tool could run over the hand as well, multiply and set the translation field and iterate over every slot... I think this would work fine? I don't see any direct issues.

**Comment by Frooxius** [comment]
Do you want this to just lengthen the bones a bit? You say that you want a wider gap between fingers, so I'm not sure if you want it to push the fingers further apart from each other horizontally, or just lengthen the fingers along their direction instead.

We could add something for that, it might be even separate general component, since HandPoser doesn't currently drive position, only the rotations.

**Comment by H3BO3** [comment]
Lengthening the finger digits along their direction, I'd say. For each digit, take its forward direction and shift the position of the subsequent digit in that direction by some scalar. A separate component could work

**Comment by GearBell** [comment]
yes, lengthening would work very well, yes please.
